src/warp.jl

```diff
@@ -1,51 +1,35 @@
-# None of these types are provided by this package, so the following line seems unclean
-#     @inline Base.getindex(A::AbstractExtrapolation, v::StaticVector) = A[convert(Tuple, v)...]
-# furthermore it would be ambiguous with getindex(::Extrapolation, xs...) after https://github.com/tlycken/Interpolations.jl/pull/141
-@inline _getindex(A, v::StaticVector) = A[convert(Tuple, v)...]
-
-warp(img::AbstractArray, args...) = warp(interpolate(img, BSpline(Linear()), OnGrid()), args...)
-
-@inline _dst_type{T<:Colorant,S}(::Type{T}, ::Type{S}) = ccolor(T, S)
-@inline _dst_type{T<:Number,S}(::Type{T}, ::Type{S}) = T
-
-function warp{T,S}(::Type{T}, img::AbstractArray{S}, args...)
-    TCol = _dst_type(T,S)
-    TNorm = eltype(TCol)
-    apad, pad = Interpolations.prefilter(TNorm, TCol, img, typeof(BSpline(Linear())), typeof(OnGrid()))
-    itp = Interpolations.BSplineInterpolation(TNorm, apad, BSpline(Linear()), OnGrid(), pad)
-    warp(itp, args...)
-end
-
-@compat const FloatLike{T<:AbstractFloat} = Union{T,AbstractGray{T}}
-@compat const FloatColorant{T<:AbstractFloat} = Colorant{T}
-
-# The default values used by extrapolation for off-domain points
-@inline _default_fill{T<:FloatLike}(::Type{T}) = convert(T, NaN)
-@inline _default_fill{T<:FloatColorant}(::Type{T}) = nan(T)
-@inline _default_fill{T}(::Type{T}) = zero(T)
-
-warp{T}(img::AbstractInterpolation{T}, tform, fill=_default_fill(T)) = warp(extrapolate(img, fill), tform)
-
 """
-    warp(img, tform) -> imgw
+    warp(img, tform, [indices], [degree = Linear()], [fill = NaN]) -> imgw
 
 Transform the coordinates of `img`, returning a new `imgw` satisfying
-`imgw[x] = img[tform(x)]`. `tform` should be defined using
-CoordinateTransformations.jl.
+`imgw[I] = img[tform(I)]`. This approach is known as backward
+mode warping. The transformation `tform` should be defined using
+[CoordinateTransformations.jl](https://github.com/FugroRoames/CoordinateTransformations.jl).
 
 # Interpolation scheme
 
-At off-grid points, `imgw` is calculated by interpolation. The default
-is linear interpolation, used when `img` is a plain array, and `NaN`
-values are used to indicate locations for which `tform(x)` was outside
-the bounds of the input `img`. For more control over the interpolation
-scheme---and how beyond-the-edge points are handled---pass it in as an
-`AbstractExtrapolation` from Interpolations.jl.
+At off-grid points, `imgw` is calculated by interpolation. The
+degree of the b-spline can be specified with the optional
+parameter `degree`, which can take the values `Linear()` or
+`Constant()`.
```

Comment:
Sure. But we don't have that implemented yet, so let's finish this so that it works for what we have.

I can go either way (restrict to `Constant` and `Linear`, or support higher orders).